clerk/clients.go

```diff
@@ -0,0 +1,49 @@
+package clerk
+
+import "fmt"
+
+type ClientsService service
+
+type ClientResponse struct {
+	Object              string  `json:"object"`
+	ID                  string  `json:"id"`
+	LastActiveSessionID *string `json:"last_active_session_id"`
+	SignInAttemptID     *string `json:"sign_in_attempt_id"`
+	SignUpAttemptID     *string `json:"sign_up_attempt_id"`
+	Ended               bool    `json:"ended"`
+}
+
+func (s *ClientsService) ListAll() ([]ClientResponse, error) {
+	clientsUrl := "clients"
+	req, _ := s.client.NewRequest("GET", clientsUrl)
+
+	var clients []ClientResponse
+	_, err := s.client.Do(req, &clients)
+	if err != nil {
+		return nil, err
+	}
+	return clients, nil
+}
+
+func (s *ClientsService) Read(clientId string) (*ClientResponse, error) {
+	clientUrl := fmt.Sprintf("clients/%v", clientId)
+	req, _ := s.client.NewRequest("GET", clientUrl)
```

Comment:
The `Client.NewRequest` can return errors if we're trying to create an invalid request, e.g. invalid request method.

Since, at this point, we are in control of what we are creating, we cannot have an error.

We could, of course, have an error check here. But as I mentioned in the [other comment](https://github.com/clerkinc/clerk-sdk-go/pull/5#discussion_r560047293), I'm not a huge fan of defensive programming inside the boundaries of our application. 

I'm actually an advocate of the (arguably) radical philosophy which says that anything that can't be tested should be removed. That's one of the reasons that I wanted to have 100% coverage on this...it doesn't help you eliminate the bugs but it helps identifying cases that can never occur.